docs/adapters/adapter_api.md

```diff
@@ -6,9 +6,65 @@ NOTE: This API is currently under active development and is highly likely to cha
 
 __TOC__
 
+## Overview
+
+An Adapter is the interface between the juttle runtime and a backend data source. The backend consists of a set of data points. Each data point contains a number of fields. In most cases, each data point also contains a timestamp.
+
+For reads, the adapter's job is to interpret the options included in the juttle `read` command into a set of matching data points, construct json objects representing those data points, and pass them to the juttle runtime. For writes, the adapter's job is to take the output of programs and "save" that output by writing data points to the backend.
+
+More sophisticated adapters work together with the juttle optimizer to push aggregation operations directly into the backend. For example, to count the number of data points in a given time period you could simply fetch all the data points and have the juttle program perform the counting. However, if supported by the backend it would be more efficient to count the number of data points directly in the backend and return the count instead.
```

Comment:
Not that I can see, I don't see any docs or anything in the wiki related to optimization.